src/Control/Inciter/InputDeck/Grammar.hpp

```diff
@@ -581,14 +590,37 @@ namespace deck {
                                     eq,
                                     param > {};
 
+  //! put in PDE parameter for equation matching keyword
+  template< typename eq, typename keyword, typename p,
+            class kw_type = tk::grm::number >
+  struct parameter :
+         tk::grm::process< use< keyword >,
+                           tk::grm::Store_back< tag::param, eq, p >,
+                           kw_type > {};
+
   //! Boundary conditions block
   template< class keyword, class eq, class param >
   struct bc :
          pegtl::if_must<
            tk::grm::readkw< typename use< keyword >::pegtl_string >,
            tk::grm::block<
              use< kw::end >,
-             tk::grm::bcparam< use, kw::p_farfield, tag::p_farfield >,
+             tk::grm::parameter_vector< use,
+                                        use< kw::sideset >,
+                                        tk::grm::Store_back_back,
+                                        tk::grm::start_vector,
+                                        tk::grm::check_vector,
+                                        eq,
+                                        param > > > {};
+
+  //! Farfield boundary conditions block
+  template< class keyword, class eq, class param >
+  struct subsonic_bc :
+         pegtl::if_must<
+           tk::grm::readkw< typename use< keyword >::pegtl_string >,
+           tk::grm::block<
+             use< kw::end >,
+             parameter< eq, kw::farfield_pressure, tag::farfield_pressure >,
```

Comment:
I have created this separate `subsonic_bc` block that takes the extra `farfield_pressure` input line, because if we had kept it inside the more generic `bc` struct above, then _all_ PDE types that use _any_ BC must have a storage for farfield, and this makes no sense for Transport. This allows having a storage for `farfield_pressure` in `CompFlowPDEParameters` and nowhere else, because `subsonic_bc` is only given in the `compflow` struct grammar block, see below.